vault/identity_store_entities.go

```diff
@@ -838,20 +834,36 @@ func (i *IdentityStore) mergeEntity(ctx context.Context, txn *memdb.Txn, toEntit
 							fmt.Errorf("mountAccessor: %s, toEntity ID: %s, fromEntity ID: %s, conflicting toEntity alias ID: %s, conflicting fromEntity alias ID: %s",
 								toAlias.MountAccessor, toEntity.ID, fromEntityID, toAlias.ID, fromAlias.ID))
 
+						var toAliasMountType string
+						var toAliasMountPath string
+						mountValidationRespToAlias := i.router.ValidateMountByAccessor(toAlias.MountAccessor)
+						if mountValidationRespToAlias != nil {
+							toAliasMountType = mountValidationRespToAlias.MountType
+							toAliasMountPath = mountValidationRespToAlias.MountPath
+						}
+
+						var fromAliasMountType string
+						var fromAliasMountPath string
+						mountValidationRespFromAlias := i.router.ValidateMountByAccessor(fromAlias.MountAccessor)
+						if mountValidationRespFromAlias != nil {
+							fromAliasMountType = mountValidationRespFromAlias.MountType
+							fromAliasMountPath = mountValidationRespFromAlias.MountPath
+						}
+
 						// Also add both to our summary of all clashes:
 						aliasesInvolvedInClashes = append(aliasesInvolvedInClashes, aliasClashInformation{
 							Entity:    toEntity.Name,
 							EntityId:  toEntity.ID,
 							Alias:     toAlias.Name,
-							Mount:     toAlias.MountType,
-							MountPath: toAlias.MountPath,
+							Mount:     toAliasMountType,
+							MountPath: toAliasMountPath,
 						})
 						aliasesInvolvedInClashes = append(aliasesInvolvedInClashes, aliasClashInformation{
 							Entity:    fromEntity.Name,
 							EntityId:  fromEntityID,
 							Alias:     fromAlias.Name,
-							Mount:     fromAlias.MountType,
-							MountPath: fromAlias.MountPath,
+							Mount:     fromAliasMountType,
+							MountPath: fromAliasMountPath,
 						})
```

Comment:
Would it make sense to abstract out a function for this so we can do:
```go
aliasesInvolvedInClashes.appendClashAliases(i.router.ValidateMountByAccessor(toAlias.MountAccessor))
aliasesInvolvedInClashes.appendClashAliases(i.router.ValidateMountByAccessor(fromAlias.MountAccessor))
```
or some such?

Reply:
I thought about that, but I don't really know what complexity we'd be saving. This function would only ever be used here, and only ever used exactly twice. I don't like that the code is a little verbose but to that end, a `?.` or similar operator is not available in Go, unfortunately.

I'm open to other thoughts, and I'm not too religious about this, but my instinct was that it didn't make the code cleaner and yeah, its only use-case is exactly twice in this function.

Reply:
Maybe we can just iterate over the two mount accessors for the sake of reducing SLOC?

Reply:
How do you mean?

Reply:
I was thinking something like this, but not sure if it's an improvement. Feel free to ignore:
```go
for _, mountAccessor := range []string{toAlias.MountAccessor, fromAlias.MountAccessor} {
							var aliasMountType string
							var aliasMountPath string
							mountValidationRespToAlias := i.router.ValidateMountByAccessor(mountAccessor)
							if mountValidationRespToAlias != nil {
								aliasMountType = mountValidationRespToAlias.MountType
								aliasMountPath = mountValidationRespToAlias.MountPath
							}
							aliasesInvolvedInClashes = append(aliasesInvolvedInClashes, aliasClashInformation{
								Entity:    toEntity.Name,
								EntityId:  toEntity.ID,
								Alias:     toAlias.Name,
								Mount:     aliasMountType,
								MountPath: aliasMountPath,
							})
```

Reply:
Yeah, I think in this case, the extra duplication is probably a lesser evil than any of the refactoring options. I think I'll merge as-is, but it was good to talk through the other options. Thanks!